packages/design-system/src/components/Accordion/AccordionItem.tsx

```diff
@@ -0,0 +1,93 @@
+import React, { FunctionComponent, useState } from 'react';
+import classNames from 'classnames';
+import uniqueId from 'lodash.uniqueid';
+
+export interface AccordionItemProps {
+  /**
+   * Class to be applied to the header `<button>` of an accordion item.
+   */
+  buttonClassName?: string;
+
+  children?: React.ReactNode;
+  /**
+   * Class to be applied to the heading `<h2>` tag of an accordion item.
+   */
+  contentClassName?: string;
+
+  /**
+   * Boolean to expand the accordion.
+   */
+  defaultOpen?: boolean;
+  /**
+   * Text for the accordion item heading.
+   */
+  heading: React.ReactNode | string;
+  /**
+   *  Heading type to override default `<h2>`.
+   */
+  headingLevel?: '1' | '2' | '3' | '4' | '5' | '6';
+  /**
+   *  If not provided, a unique id will be automatically generated and used.
+   */
+  id?: string;
+}
+export interface AccordionItemState {
+  isOpen: boolean;
+}
+export class AccordionItem extends React.Component<
+  AccordionItemProps, 
+  AccordionItemState
+> {
+  contentId: string;
+  buttonId: string;
+
+  constructor(props: AccordionItemProps) {
+    super(props);
+
+    this.state = { isOpen: !!props.defaultOpen };
+    this.onClick = this.onClick.bind(this);
+    this.contentId = props.id || uniqueId('accordionItem_');
+    this.buttonId = `${this.contentId}-button`;
+  }
+  // Set the state for opening and closing an accordion item
+  onClick(): void {
+    this.setState({ isOpen: !this.state.isOpen });
+  }
+
+  render(): React.ReactNode {
+    const {
+      buttonClassName,
+      children,
+      contentClassName,
+      heading,
+      headingLevel = '2',
```

Comment:
The docs site shows a headingLevel as required, but I do not get any error if I do not include it. It just defaults to 2. Also the doc site doesn't show anything in the default column.

If we are going to default to 2, shouldn't we put that in the default column and remove the required tag?

Reply:
One other thing, no matter what headLevel I pass in, I see that `<h1>` `<h2>` etc gets put in the code, but visually there is no difference. Is this how it should be?

Reply:
> The docs site shows a headingLevel as required, but I do not get any error if I do not include it. It just defaults to 2. Also the doc site doesn't show anything in the default column.

I don't see headingLevel as required here http://design-system-demo.s3-website-us-east-1.amazonaws.com/WNMGDS-41-Accordion/components/accordion/ but you are right it's not showing as a default of 2 I think this is an issue with our prop autogenerated docs @nichia do you know anything about that?


>One other thing, no matter what headLevel I pass in, I see that <h1> <h2> etc gets put in the code, but visually there is no difference. Is this how it should be?

That is working as expected and it's more for semantics

Reply:
Thanks!